app/models/portfolio.rb

```diff
@@ -7,6 +7,8 @@ class Portfolio < ApplicationRecord
 
   destroy_dependencies :portfolio_items
 
+  before_save { self.statistics = stats }
```

Comment:
Not sure if `before_save` is need, but was considering at least a `before_create` so the stats are present in `metadata` when the initial Portfolio is created.  Keep this to get feedback.  I think of it as a "true-up" operation for the stats when other actions update the Portfolio object.